### DISCUSSION TITLE
Panel prepareFields can cause the disappearance of fields when the numeric keys are mismatched due to array_filter

### DISCUSSION THREAD

**Discussion by Tarpsvo** [laravel/nova-issues#5287] [Q&A] [answered]
- Laravel Version: 9.48
- Nova Version: 4.21.0
- PHP Version: 8.1

### Description:

When you have conditional fields inside a Panel (ie `array_filter([somethingIsEnabled() ? Text::make('Whatever') : null]`), the panel can cause the disappearance of multiple fields as the numeric keys will become messed up and merging will break.

### Detailed steps to reproduce the issue on a fresh Nova installation:

Create a resource with the following fields:
```
 public function fields(NovaRequest $request)
{
    return [
        Text::make('Name'),

        Panel::make('Twost', array_filter([
            null,
            Text::make('Content'),
        ])),

        Text::make('Where have I gone?'),
    ];
}
```

And this will happen:
![image](https://user-images.githubusercontent.com/2018660/215763223-06bd1d7d-c74c-4db4-8dbc-8884900c0111.png)

Remove the `null` from inside the panel and this will show:
![image](https://user-images.githubusercontent.com/2018660/215763355-cf112d5c-1903-4b01-9133-b60331b481e5.png)

### Proposed fix

I suggest using `->values()->toArray()` instead of `->all()` inside Panel's `prepareFields()` function. I would even go as far as to call `->filter()` on the field collection before the rejection statement.

**Comment by crynobone** [comment]
We have no plan to support `null` as valid options, as for now you should be able to do the following:

```php
Panel::make('Twost', collect([
    somethingIsEnabled() ? Text::make('Whatever') : null,
    Text::make('Content'),
])->values()->all()),
```

**Comment by crynobone** [comment, marked as answer]
In 4.22.0 you can do the following:

```php
Panel::make('Twost', [
    $this->mergeWhen(somethingIsEnabled(), [
        Text::make('Whatever'),
    ]),
    Text::make('Content'),
]),
```
